### ISSUE TITLE
clear scrollback

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#421]
```
it would be nice to have an option clear scrollback
```

Original issue reported on code.google.com by `Torrenti...@gmail.com` on 27 Feb 2014 at 8:10

**Comment by mintty** [comment]
unclear (option by escape sequence? menu function?) and not supported by other terminals
Closing for now.

**Comment by NoAnyLove** [comment]
It would be nice to have a menu function to clear the scrollback buffer.

The escape sequence `echo -e "\033c"`, found at <http://cygwin.com/ml/cygwin/2009-03/msg00495.html>, also clears the background color if using base16-shell, which is not what I expect.

**Comment by mintty** [comment]
I take it as a request to have that function in the context menu. Like another request (#539) this is likely to become available once there is a configurable context menu. I do not wish to clutter the default menu with too many entries.

What is "base16-shell"?
What's wrong about ESC c?

**Comment by NoAnyLove** [comment]
[base16-shell](https://github.com/chriskempson/base16-shell) is a shell script to change shell's default ANSI colors. I use it to change the color theme of MinTTY, it's easier than modifying the .minttyrc.

The problem is, if I use `echo -e "\033c"` to clear the scrollback buffer, the background color will restore to black. I tried `echo -e "\033c"` in gnome-terminal, the background color won't change.

**Comment by mintty** [comment]
Both ESC c (Hard Terminal Reset, RIS) and ESC [ ! p (Soft Terminal Reset, DECSTR, not implemented by mintty) clear the background colour in xterm, which is compliant with the DEC terminal sequences (http://vt100.net/docs/vt220-rm/chapter4.html#S4.18), so gnome-terminal does not strictly do the proper thing. Arguably, at least one of the sequences could be tweaked to do what you want. Independently of that an optional context menu entry as you requested would be OK. There isn't simply the mechanism available yet to configure the context menu.

**Comment by NoAnyLove** [comment]
Understood, thanks.

BTW, is there any plan on the optianl context menu? I mean, if there is a schedule to implement that?

**Comment by mintty** [comment]
No, this is not a scheduled project.

**Comment by NoAnyLove** [comment]
OK.

Could you please tell me the sequence that can clear the scrollback while not change the background color, if possible. Thanks.

**Comment by mintty** [comment]
Actually, there is a sequence to clear the scrollback buffer which I just looked up myself in http://invisible-island.net/xterm/ctlseqs/ctlseqs.txt:
```
echo -e "\e[3J"
```

**Comment by mintty** [comment]
Released extended context menu function with 2.7.5.
